### ISSUE TITLE
[Bug]

### ISSUE THREAD

**Issue by alanrenouf** [tomaae/homeassistant-portainer#17]
Setup the HACS integration and registered my portainer but get the following error:

![image](https://github.com/tomaae/homeassistant-portainer/assets/4140758/0582e497-0c8d-4c93-804e-f558ef566160)

 - Home Assistant version: HA 2023.9.3
 - Portainer integration version: v1.0.2
 - Portainer software version: 2.19.1

**Comment by tomaae** [comment]
check your ha log for errors

**Comment by alanrenouf** [comment]
Logger: homeassistant.config_entries
Source: config_entries.py:1248
First occurred: 2:36:27 PM (1 occurrences)
Last logged: 2:36:27 PM

Config entry 'Portainer' for portainer integration not ready yet: list index out of range; Retrying in background

**Comment by tomaae** [comment]
is that only error message?
if so, remove the configuration entry, enable debug and readd it again so we get some information.

**Comment by alanrenouf** [comment]
Thanks for the help, here is the log file once i enabled debug.

[home-assistant_2023-09-27T19-13-02.842Z.log](https://github.com/tomaae/homeassistant-portainer/files/12742502/home-assistant_2023-09-27T19-13-02.842Z.log)

**Comment by tomaae** [comment]
Hmm, I can see integration itself is gathering data correctly, but there are bunch of crashes related to different integrations and such.
Try to fix those too, so we know its not interfering.
Also, have you tried removing and readding the portainer?

**Comment by alanrenouf** [comment]
The other integrations are expected, I was just trying some things out, I removed them so here is another log file, i tried completely removing and re-adding the plugin but still doesnt work

[home-assistant_2023-09-28T21-23-23.599Z.log](https://github.com/tomaae/homeassistant-portainer/files/12754110/home-assistant_2023-09-28T21-23-23.599Z.log)

**Comment by tomaae** [comment]
I see, you are not running it on docker, seems to be kubernates?
I dont have any real experience with that. Can you give me step by step info on how to deploy same type of envinroment you have? I can spin up a VM for it.

**Comment by gapthorpe** [comment]
I had the same error when I had two environments in Portainer, one being a swarm externally (which actually was empty). Removing this environment, the integration started working fine for the local one. 

Thanks!

**Comment by tomaae** [comment]
> I had the same error when I had two environments in Portainer, one being a swarm externally (which actually was empty). Removing this environment, the integration started working fine for the local one.
> 
> Thanks!

yea, its could be the same issue. but its better to open issue when you have some problem like that.
integration is still in baby stages, so there are many outlier cases that need an workaround.

**Comment by gapthorpe** [comment]
Understood - it was a half finished config from a failed experiment, so was meaning to delete and just saw this same error before I did. Thought it may help if @alanrenouf has any empty or failing environments

**Comment by alanrenouf** [comment]
> Understood - it was a half finished config from a failed experiment, so was meaning to delete and just saw this same error before I did. Thought it may help if @alanrenouf has any empty or failing environments

Thanks @gapthorpe appreciate the information, I only have 1 environment but this definitely looks like some response the API is giving that the code isn't expecting. @tomaae is there any way to enable debug on the GET requests?

**Comment by tomaae** [comment]
yes, but that is not your issue.
integration was build for docker. you use kubernates which is handled differently by portainer. I need help setting up development environment for kubernates to implement it.

**Comment by alanrenouf** [comment]
@tomaae Probably best to follow something like this for a quick start: https://yashsrivastav.hashnode.dev/getting-started-with-portainer-using-kind

**Comment by alanrenouf** [comment]
Anything I can do to help with this one?

**Comment by tomaae** [comment]
not really, I just have very little time right now due to family issues.
update for this one will probably take a bit, since this integration needs more features to be really useful and of course kubernates support.

**Comment by SphtKr** [comment]
Another with this issue. I'm running Kubernetes via microk8s on Ubuntu (Jammy)...this should be probably one of the easiest environments to build a test VM for.

Install Jammy, install microk8s (via snap) and then install the portainer add-on with `microk8s enable portainer`, which should give you Portainer available on port 30777. I actually run HA inside microk8s, so I'm trying to access it via `portainer.portainer.svc.cluster.local:9000` , but it should work from outside the cluster at port 30777.

**Comment by jasonwitty** [comment]
@tomaae you can setup k3s on a raspberry pi for testing this, i am also available to help if needed. let me know. thanks.

**Comment by fuzzysb** [comment]
I have the same issue, i have a Portainer instance with a 9 node MicroK8s cluster and an additional 3 docker hosts and i receive this same error.  happy to be a tester for any code changes

**Comment by padraignix** [comment]
Ran into the same issue when trying to set this up in HA. Have a docker setup in portainer, along with a k3s cluster.

**Comment by broyuken** [comment]
still having this issue in 2025
